crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -197,7 +203,8 @@ impl SettlementClient for StarknetSettlementClient {
         if block_number.is_empty() {
             return Err(eyre!("Could not fetch last block number from core contract."));
         }
-        Ok(block_number[0].try_into()?)
+
+        Ok(u64::from_le_bytes(block_number[0].to_bytes_le()[0..8].try_into().unwrap()))
```

Comment:
can you add the test case for this which we had yesterday

Reply:
Added !!